src/socket.c

```diff
@@ -388,7 +388,7 @@ void pthreads_socket_connect(zval *object, int argc, zend_string *address, zend_
 			struct sockaddr_in sin = {0};
 
 			if (argc != 2) {
-				zend_throw_exception_ex(spl_ce_RuntimeException, 0, "Socket of type AF_INET requires 2 arguments");
+				zend_throw_exception_ex(spl_ce_InvalidArgumentException, 0, "Socket of type AF_INET requires 2 arguments");
```

Comment:
This maybe ought to be `ArgumentCountError` instead of `InvalidArgumentException`, but `WRONG_PARAM_COUNT` only throws exceptions in strict mode - not sure how to deal with this